rkt/metadata_service.go

```diff
@@ -75,6 +71,109 @@ func init() {
 	cmdMetadataService.Flags.BoolVar(&flagNoIdle, "no-idle", false, "exit when last container is unregistered")
 }
 
+type mdsPod struct {
+	uuid     types.UUID
+	ip       string
+	manifest *schema.PodManifest
+	apps     map[string]*schema.ImageManifest
+}
+
+type podStore struct {
+	byIP   map[string]*mdsPod
+	byUUID map[types.UUID]*mdsPod
+	mutex  sync.Mutex
```

Comment:
I think it's a much clearer indication that the entire struct is not threadsafe. If you want to keep it as a named member then move it above the others because by convention the things the mutex protects are listed below it